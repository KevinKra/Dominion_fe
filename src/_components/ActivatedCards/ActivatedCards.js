import React from "react";
import Card from "../Card/Card";
import { connect } from "react-redux";
import * as SVGLoaders from "svg-loaders-react";
import "./ActivatedCards.scss";

export function ActivatedCards(props) {
  const turnNotifier = props.playerTurn.isActive ? (
    <p
      className={
        props.activatedCards.length === 0 ? "turn-notifier" : "turn-notifier-void"
      }
    >
      It's Your Turn!
    </p>
  ) : (
    <div className='waiting-response'>
      <p className='turn-notifier'>Waiting For Opponent</p>
      <SVGLoaders.ThreeDots className='turn-loader' />
    </div>
  );
  return (
    <section className='ActivatedCards'>
      {turnNotifier}
      {props.activatedCards.map(card => {
        return (
          <Card
            name={card.name}
            desc={card.desc}
            category={card.category}
            image={card.image}
            cost={card.cost}
            id={card.id}
            key={card.id}
          />
        );
      })}
    </section>
  );
}

export const mapStateToProps = state => ({
  playerTurn: state.playerTurn,
  activatedCards: state.activatedCards
});

export default connect(mapStateToProps)(ActivatedCards);
