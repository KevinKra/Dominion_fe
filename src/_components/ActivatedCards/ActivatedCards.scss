@import '../../styles/_variables.scss';
@import '../../styles/_animations.scss';

.ActivatedCards {
  display: flex;
  position: absolute;
  justify-content: center;
  font-family: 'Germania One', cursive;
  z-index: 1;
  width: 50%;
  margin-top: 0.5rem;
  border: 1px solid $color2;
  border-bottom: 1px dashed $color1;
  border-top: 1px dashed $color1;
  background-color: rgba(0, 0, 0, 0.55);
  min-height: 100px;
  transition: 1s all;
  .waiting-response {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
  }
  .turn-notifier {
    color: $color1;
    font-size: 22px;
    margin: auto;
    animation: fadeIn 0.7s;
  }
  .turn-notifier-void {
    display: none;
  }
  .turn-loader {
    // border: 1px solid red;
    margin: 0.5rem;
    transform: scale(0.5);
  }
}
