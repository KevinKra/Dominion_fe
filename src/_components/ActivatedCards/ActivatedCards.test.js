import React from "react";
import ActivatedCards from "./ActivatedCards";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("<ActivatedCards />", () => {
  let wrapper = shallow(<ActivatedCards />);
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
