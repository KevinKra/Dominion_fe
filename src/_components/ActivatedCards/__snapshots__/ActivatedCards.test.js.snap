// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivatedCards /> should match the snapshot 1`] = `
<section
  className="ActivatedCards"
>
  ActivatedCards
</section>
`;
